oecophylla/taxonomy/taxonomy.rule

```diff
@@ -123,6 +123,62 @@ rule taxonomy_kraken:
                   """)
 
 
+rule taxonomy_bracken:
+    """
+    Runs Kraken plus Bracken using general defaults.
+    """
+    input:
+        forward = qc_dir + "{sample}/filtered/{sample}.R1.trimmed.filtered.fastq.gz",
+        reverse = qc_dir + "{sample}/filtered/{sample}.R2.trimmed.filtered.fastq.gz"
+    output:
+        table = taxonomy_dir + "{sample}/bracken/{sample}_table.txt",
+        report = taxonomy_dir + "{sample}/bracken/{sample}_report.txt"
+    params:
+        kraken_env = config['envs']['kraken'],
+        kraken_db = config['params']['kraken']['kraken_db'],
+        bracken_db = config['params']['kraken']['bracken_db']
+    threads:
+        12
+    log:
+        taxonomy_dir + "logs/taxonomy_bracken.sample=[{sample}].log"
+    benchmark:
+        "benchmarks/taxonomy/taxonomy_bracken.sample=[{sample}].txt"
+    run:
+        with tempfile.TemporaryDirectory(dir=find_local_scratch(TMP_DIR_ROOT)) as temp_dir:
+            table_base = os.path.basename(output['table'])
+            report_base = os.path.basename(output['report'])
+
+            shell("""
+                  set +u; {params.kraken_env}; set -u
+
+                  kraken {input.forward} {input.reverse} \
+                    --db {params.kraken_db} \
+                    --paired \
+                    --fastq-input \
+                    --gzip-compressed \
+                    --only-classified-output \
+                    --threads {threads} \
+                    1> {temp_dir}/map.tmp \
+                    2> {log}
+
+                  kraken-report {temp_dir}/map.tmp \
+                    --db {params.kraken_db} \
+                    1> {temp_dir}/report.tmp \
+                    2>> {log}
```

Comment:
not a distinct log?

Reply:
The three steps (kraken, kraken-report and est_abundance.py) are highly uneven in computational expensive. kraken consumes most time. kraken-report is but an output parser. est_abundance.py (Bracken) is close to a parser as well. I think it is not necessary to assign one log file per step.